src/utils/NetworkUtils.ts

```diff
@@ -123,6 +123,56 @@ export function chainIsL1(chainId: number): boolean {
   return [CHAIN_IDs.MAINNET, CHAIN_IDs.SEPOLIA].includes(chainId);
 }
 
+/**
+ * Determines whether a chain ID runs on an EVM-like execution layer.
+ * @param chainId Chain ID to evaluate.
+ * @returns True if chain corresponding to chainId has an EVM-like execution layer.
+ */
+export function chainIsEvm(chainId: number): boolean {
```

Comment:
Thoughts on moving this up to constants? So like `ChainFamily`, add another parameter to the `PublicNetwork` interface like `executionLayer` and specify evm/svm/etc.

Reply:
Yeah I think we need some central way of inferring the execution environment. We can infer most of it based on the ChainFamily alone, though that doesn't help with Ethereum, Linea and Scroll. It'd be nice if we could avoid adding another property, but we might be forced to.